src/lib/s2k.c

```diff
@@ -85,7 +85,7 @@ pgp_s2k_round_iterations(size_t iterations)
 uint8_t
 pgp_s2k_encode_iterations(size_t iterations)
 {
-    for (uint8_t c = 0; c <= 255; ++c) {
+    for (uint16_t c = 0; c < 256; ++c) {
```

Comment:
@MohitKumarAgniotri could we add some documentation referencing the RFC here?

e.g,. some part of the following
> For compatibility, when an S2K specifier is used, the special value
> 254 or 255 is stored in the position where the hash algorithm octet
> would have been in the old data structure.  This is then followed
> immediately by a one-octet algorithm identifier, and then by the S2K
> specifier as encoded above.
> 
> Therefore, preceding the secret data there will be one of these
> possibilities:
> 
>     0:           secret data is unencrypted (no passphrase)
>     255 or 254:  followed by algorithm octet and S2K specifier
>     Cipher alg:  use Simple S2K algorithm using MD5 hash
> 
> This last possibility, the cipher algorithm number with an implicit
> use of MD5 and IDEA, is provided for backward compatibility; it MAY
> be understood, but SHOULD NOT be generated, and is deprecated.